lfucg-exactions/server/accounts/serializers.py

```diff
@@ -26,15 +26,20 @@ class AccountSerializer(serializers.ModelSerializer):
     balance = serializers.SerializerMethodField(read_only=True)
     credit_availability = serializers.SerializerMethodField(read_only=True)
 
+    def _get_account_balance(self, obj):
+        _account_balance = calculate_account_balance(obj.id)
```

Comment:
You need to save the calculated results so that you don't recalculate it. What you're doing here still runs twice.